WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -96,8 +96,8 @@ def self.status_sort_rank(role)
       # operator to access the parameters.
       (
         (status.present? && status != (is_actual_role ? role.metadata.status : role[:metadata][:status])) ||
-        (is_active.present? && is_active != (is_actual_role ? role.is_active? : role[:is_active])) ||
-        (is_group_hidden.present? && is_group_hidden != (is_actual_role ? role.group.is_hidden : role[:group][:is_hidden]))
+        (!is_active.nil? && is_active != (is_actual_role ? role.is_active? : role[:is_active])) ||
+        (!is_group_hidden.nil? && is_group_hidden != (is_actual_role ? role.group.is_hidden : role[:group][:is_hidden]))
```

Comment:
Why are you changing this from `present?` to `!nil`?
And why are you changing it only for `is_active` and `is_group_hidden`, but not for `status` one line above?

Reply:
I need to evaluate the expression to the right side of `&&` only if the corresponding parameter (status, is_active, is_group_hidden) is present. If status is present, then it's going to be a string so `status.present?` will be `true` always. But other two parameters are boolean, hence can be `false` as well. Unfortunately rails is returning `false` for `x.present?` if `x` is `false`.